docs/DeveloperGuide.md

```diff
@@ -1476,3 +1476,64 @@ If you wish to add new checks, simply add the check file with a filename `check-
         1.  `search`
         2.  `search -g musik` In this case it would attempt to search for `musik`, and return no results.
         3.  `search -g`
+<br/>
+  
+### D.x: Create a new Workspace
+1.  Create a new Workspace with `workspace` feature
+    1.  Prerequisite:  None.
+    
+    2.  Test case: `workspace -n Crunchyroll` <br/>
+    Expected: Workspace will be successfully created with success message.
+    
+    3.  Test case: `workspace -n Crunchyroll__` <br/>
+    Expected: Workspace creation will fail with error message stating names must be alphanumeric and spaces only.
+    
+        1.  Other incorrect commands to try: 
+            1.  `workspace`
+            2.  `workspace -n `
+            3.  `workspace -N`
+<br/>
+            
+### D.x: Switch to a different Workspace
+1.  Switch to a different Workspace with `workspace` feature
+    1.  Prerequisite:  Workspace must first exist before switching to them. In this example we presume Workspace named `Default` and `CrunchyOreo` exists and currently active Workspace is the former.
+    
+    2.  Test case: `workspace -s CrunchyOreo` <br/>
+    Expected: Workspace will be successfully switched from `Default` to `CrunchyOreo`.
+    
+    3.  Test case: `workspace -s CrunchyOreo__` <br/>
+    Expected: Workspace switch will fail given how the name request does not meet requirement of being alphanumeric and spaces only.
+    
+        1.  Other incorrect commands to try: 
+            1.  `workspace`
+            2.  `workspace -s `
+            3.  `workspace -S`
+<br/>
+
+### D.x: List all Workspace
+1.  List all Workspace with `workspace` feature
+    1.  Prerequisite:  Workspace must first exist to be able to list them. In this example we presume Workspace named `Default` and `CrunchyOreo` exists and currently active Workspace is the former.
+    
+    2.  Test case: `workspace -l` <br/>
+    Expected: Workspace `Default` and `CrunchyOreo` will be listed.
+    
+        1.  Other incorrect commands to try: 
+            1.  `workspace`
+<br/>
+            
+### D.x: Delete a Workspace
+1.  Delete a Workspace with `workspace` feature
+    1.  Prerequisite:  Workspace must first exist before deleting them. In this example we presume Workspace named `Default` and `CrunchyOreo` exists and currently active Workspace is the latter.
+    
+    2.  Test case: `workspace -d Default` <br/>
+    Expected: Workspace will be successfully deleted.
+    
+    3.  Test case: `workspace -d default` <br/>
+    Expected: Workspace deletion will fail given default does not exists.
+    
+        1.  Other incorrect commands to try: 
+            1.  `workspace`
+            2.  `workspace -d `
+            3.  `workspace -d`
```

Comment:
I understand that you are trying to emphasise on the space for the 2nd one, but in pdf format I feel that it may not be very apparent. Maybe just keep one of them?

Reply:
Alright thank you, I have updated accordingly!